waku/v2/peermanager/peer_connector.go

```diff
@@ -272,6 +272,11 @@ func (c *PeerConnectionStrategy) dialPeers() {
 	}
 }
 
+type DialError struct {
```

Comment:
Better to place this type in utils package so its accessible from other places.

This file mainly dials/connects to peers for relay and when the peers are added to peerstore. 
we need to capture on-demand dialing scenarios especially for req/resp protocols such as  lightpush/filter/store e.g https://github.com/waku-org/go-waku/blob/280039120431f7085a9bd5beb6d9804889cf3d3f/waku/v2/protocol/lightpush/waku_lightpush.go#L196

So, my recommendation would be to search the code for NewStream or Connect methods and look for dial errors and report them in the same way.

i am not too sure, but we may have to check in status-go code as well because some peers are dialed directly from there (which i think are mostly fleet peers). @richard-ramos can confirm.

Reply:
Refactored `DialError` into utils and added a public function to `PeerConnector` to emit the error. Called both in `PeerConnectionStrategy.dialPeer` and `WakuNode.connect`

Reply:
> Connections are now done via `AddDiscoveredPeer`

that is when a peer is getting added. Now, consider the peer was discovered and added in lightclient but not used so the connection may get dropped due to being idle. And when the client wants to send a msg via lightpush a new peer or the same peer may get selected and new connection will be established via NewStream call. isn't it? unless i am missing something.

Reply:
I took a look at go-libp2p and it does seem to call `host.connect` in `NewStream`, at least here: https://github.com/libp2p/go-libp2p/blob/9038a72c4863778e6a26814aab201cee6d93d640/p2p/host/basic/basic_host.go#L689

Does it make sense to use this same logic for handling dial errors in every call to `NewStream` in go-waku? https://github.com/waku-org/go-waku/blob/4bba45617a48d3e3653a0fcce2641a65d2d5b79f/waku/v2/peermanager/peer_connector.go#L295

Reply:
i think so, because stream creation errors would mostly be due to underlying connectivity issues only unless there is a bug in stream management which i think is rare.

Reply:
I moved the function to `PeerManager` and added a call to it in every place where `NewStream` is used